File: leetcode-submissions/713_Subarray_Product_Less_Than_K.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numSubarrayProductLessThanK = function(nums, k) {
    let start=0;
    let end=0;
    let product=1;
    let count=0;
    while(end<nums.length){
        product*=nums[end];
        end++;
        while(start<nums.length && product>=k){
            product/=nums[start];
            start++;
        }
        if(product<k){
            count+=(end-start+1);
        }
        return count;
    }
};

File: leetcode-submissions/713_Subarray_Product_Less_Than_K.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        //now solving this question by using the sliding window approach
        int count=0;
        for(int i=0;i<nums.size();i++){
            int product=1;

            for(int j=i;j<nums.size();j++){
                product*=nums[j];
                if(product<k){
                    count++;
                }

            }
        }
        return count;
    }
};